### PR TITLE
[ContactStructuralMechanicsApplication] Updating tests

### PR DESCRIPTION
**Pull request by loumalouomega**
This is the part  3 of a series of updates in the ContactStructuralMechanicsApplication

I don't recommend to check the changes, basically tests files updated. The relevant content is the one from #1639 and #1640

### CODE REVIEW

**Comment by loumalouomega** [comment]
Requires #1639 and #1640 to work

**Comment by loumalouomega** [comment]
Using the default parameters

**Comment by loumalouomega** [comment]
@RiccardoRossi

File: applications/ContactStructuralMechanicsApplication/python_scripts/alm_contact_process.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def __init__(self,model_part,params):
             "assume_master_slave"         : "",
             "contact_type"                : "Frictionless",
             "frictional_law"              : "Coulomb",
-            "search_factor"               : 2.0,
+            "search_factor"               : 3.5,
```
Comment (RiccardoRossi) [inline applications/ContactStructuralMechanicsApplication/python_scripts/alm_contact_process.py]: we said to remove this, didn't we?

File: applications/ContactStructuralMechanicsApplication/python_scripts/alm_contact_process.py [hunk 2]
```diff
@@ -43,7 +43,8 @@ def __init__(self,model_part,params):
             "type_search"                 : "InRadius",
             "check_gap"                   : "CheckMapping",
             "database_step_update"        : 1,
-            "integration_order"           : 3,
+            "integration_order"           : 2,
```
Comment (RiccardoRossi) [inline applications/ContactStructuralMechanicsApplication/python_scripts/alm_contact_process.py]: why don't you leave the defaults? (that is, you take out the line)

File: applications/ContactStructuralMechanicsApplication/tests/ALM_frictionless_contact_test_2D/mesh_moving_notmatching_test_parameters.json [hunk 3]
```diff
@@ -147,20 +146,20 @@
         "process_name"  : "ALMContactProcess",
         "Parameters"            : {
             "model_part_name"             : "Structure",
-            "assume_master_slave"         : "Parts_Parts_Auto1",
+            "assume_master_slave"         : "Parts_Parts_Auto2",
             "contact_model_part"          : "CONTACT_Contact_Auto1",
             "contact_type"                : "Frictionless",
-            "search_factor"               : 2.0,
+            "search_factor"               : 3.5,
```
Comment (RiccardoRossi) [inline applications/ContactStructuralMechanicsApplication/tests/ALM_frictionless_contact_test_2D/mesh_moving_notmatching_test_parameters.json:152]: i would remove this line. This way the default will be used, and if you change the default all cases will be affected at once

File: applications/ContactStructuralMechanicsApplication/tests/ALM_frictionless_contact_test_2D/mesh_moving_notmatching_test_parameters.json [hunk 4]
```diff
@@ -147,20 +146,20 @@
...
             "integration_order"           : 2,
-            "active_check_factor"         : 1.0e-1,
+            "active_check_factor"         : 1.0e-2,
             "normal_variation"            : "NO_DERIVATIVES_COMPUTATION",
             "stiffness_factor"            : 10.0,
             "penalty_scale_factor"        : 1.0,
             "manual_ALM"                  : false,
-            "penalty"                     : 1.0e6,
-            "scale_factor"                : 1.0e6,
+            "penalty"                     : 1.0e-6,
```
Comment (RiccardoRossi) [inline applications/ContactStructuralMechanicsApplication/tests/ALM_frictionless_contact_test_2D/mesh_moving_notmatching_test_parameters.json:161]: default

File: applications/ContactStructuralMechanicsApplication/tests/ALM_frictionless_contact_test_2D/simple_hertz_sphere_plate_test_parameters.json [hunk 5]
```diff
@@ -94,18 +94,7 @@
...
-            "integration_order"           : 2,
-            "active_check_factor"         : 4.0e-02,
-            "normal_variation"            : "NO_DERIVATIVES_COMPUTATION",
-            "stiffness_factor"            : 10.0,
-            "penalty_scale_factor"        : 1.0,
-            "manual_ALM"                  : true,
-            "penalty"                     : 1.0e6,
-            "scale_factor"                : 1.0e0,
-            "debug_mode"                  : false
+            "active_check_factor"         : 4.0e-02
```
Comment (RiccardoRossi) [inline applications/ContactStructuralMechanicsApplication/tests/ALM_frictionless_contact_test_2D/simple_hertz_sphere_plate_test_parameters.json:97]: in the other cases this is 1. why don't you use the default?

File: applications/ContactStructuralMechanicsApplication/tests/cpp_tests/test_integration.cpp [hunk 6]
```diff
@@ -49,7 +49,7 @@ namespace Kratos
          * Checks mass matrix computed
```
Comment (RiccardoRossi) [inline applications/ContactStructuralMechanicsApplication/tests/cpp_tests/test_integration.cpp:49]: shouldn't this be in the core?

File: applications/ContactStructuralMechanicsApplication/tests/ALM_frictionless_contact_test_2D/mesh_moving_matching_test_parameters.json [hunk 7]
```diff
@@ -151,17 +150,8 @@
             "contact_model_part"          : "CONTACT_Contact_Auto1",
             "contact_type"                : "Frictionless",
             "search_factor"               : 1.90,
-            "max_number_results"          : 1000,
-            "type_search"                 : "InRadius",
-            "integration_order"           : 2,
             "active_check_factor"         : 1.0e-1,
```
Comment (RiccardoRossi) [inline applications/ContactStructuralMechanicsApplication/tests/ALM_frictionless_contact_test_2D/mesh_moving_matching_test_parameters.json:153]: ?

File: applications/ContactStructuralMechanicsApplication/tests/ALM_frictionless_contact_test_2D/mesh_moving_matching_test_parameters.json [hunk 8]
```diff
@@ -151,17 +150,8 @@
...
-            "normal_variation"            : "NO_DERIVATIVES_COMPUTATION",
-            "stiffness_factor"            : 10.0,
-            "penalty_scale_factor"        : 1.0,
-            "manual_ALM"                  : false,
-            "penalty"                     : 2.0e4,
-            "scale_factor"                : 1.0e0,
-            "debug_mode"                  : false
+            "stiffness_factor"            : 10.0
```
Comment (RiccardoRossi) [inline applications/ContactStructuralMechanicsApplication/tests/ALM_frictionless_contact_test_2D/mesh_moving_matching_test_parameters.json:154]: ?

**Review by RiccardoRossi — APPROVED** [review]

**Pull request merged**